ibtk/include/ibtk/FEDataManager.h

```diff
@@ -619,27 +619,14 @@ class FEDataManager : public SAMRAI::tbox::Serializable
     SAMRAI::tbox::Pointer<SAMRAI::hier::PatchHierarchy<NDIM> > getPatchHierarchy() const;
 
     /*!
-     * \brief Reset range of patch levels over which operations occur.
-     *
-     * The levels must exist in the hierarchy or an assertion failure will
-     * result.
-     */
-    void setPatchLevels(int coarsest_ln, int finest_ln);
-
-    /*!
-     * \brief Get the range of patch levels used by this object.
-     *
-     * \note Returns [coarsest_ln,finest_ln+1).
+     * Get the coarsest patch level number on which elements are assigned.
      */
-    std::pair<int, int> getPatchLevels() const;
-
-    //\}
+    int getCoarsestPatchLevelNumber() const;
```

Comment:
This seems fine, although
```suggestion
    std::pait<int,int> getPatchLevelNumberRange() const;
```
also could work?

Reply:
I guess it is more clear to have the separate functions, since when you get a pair, is `second` the finest level number, or one past the finest level number, or ... ? 😀

Reply:
They ultimately do the same thing (I think I would prefer to use one-past-the-end values everywhere in the library but we don't) - I went with this interface since it is more like what all the SAMRAI objects offer and it uses the word patch to be clear that we are not talking about libMesh FE levels.

Reply:
👍

Reply:
I _think_ that one change in SAMRAI 3/4 is that they consistently use "one-past-the-end" semantics.